website/src/form-context.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { createContext, useContext, useState } from "react"
+import {
+  unstable_FormStateReturn as FormStateReturn,
+  unstable_useFormState as useFormState,
+} from "reakit/Form"
+import * as Dailp from "./graphql/dailp"
+import { usePreferences } from "./preferences-context"
+
+type FormContextType = {
+  form: FormStateReturn<any | undefined>
+  isEditing: boolean
+  setIsEditing: (bool: boolean) => void
+}
+
+const FormContext = createContext<FormContextType>({} as FormContextType)
+
+// Instantiates a form state used to keep track of the current word and information about all its features.
+export const FormProvider = (props: { children: any }) => {
+  const [isEditing, setIsEditing] = useState(false)
+  const word: Dailp.FormFieldsFragment = {} as Dailp.FormFieldsFragment
+
+  const [updateWordResult, updateWord] = Dailp.useUpdateWordMutation()
+
+  const { cherokeeRepresentation } = usePreferences()
+
+  /** Calls the backend GraphQL mutation to update a word. */
+  const runUpdate = async (variables: { word: Dailp.AnnotatedFormUpdate }) => {
+    await updateWord(variables)
+  }
+
+  const form = useFormState({
+    values: {
+      word,
+    },
+    onValidate: (values) => {
+      if (!values || !values.word) {
+        throw { values: "No word found" }
+      }
+    },
+    onSubmit: (values) => {
+      if (cherokeeRepresentation === Dailp.CherokeeOrthography.Taoc) {
+        setIsEditing(false)
+
+        // Create an array of MorphemeSegmentUpdate type to send to the backend.
+        const updatedSegments: Array<Dailp.MorphemeSegmentUpdate> = values.word[
+          "segments"
+        ].map((segment) => {
+          return {
+            morpheme: segment.morpheme,
+            gloss: segment.gloss,
+            role: segment.role,
+          }
+        })
```

Comment:
why is the `.map` needed? `segment` has too many fields, is that right?

You can also do `.map((segment) => ({ morpheme: segment.morpheme, ... }))`

Reply:
Yes, I used `.map` here since `segment` has too many fields, and I needed to convert it into a `MorphemeSegmentUpdate` object type.

Does this change look right?

<img width="664" alt="image" src="https://user-images.githubusercontent.com/40529597/189415503-6bcb5024-ecc1-4b3f-b068-8a8939f5bf22.png">